docs/proposals/2019-07-12_Image_Focal_Point_Proposal.md

```diff
@@ -0,0 +1,255 @@
+			"galleries": [{
+				"headlines": {
+					"basic": "Test with new Upload photos featuring guitars"
+				},
+				"_id": "RVOTZ67DHRCX3OKK775RKGCLPI"
+			}],
+			"galleryOrder": 12,
+			"ingestionMethod": "manual",
+			"keywords": [""],
+			"mime_type": "image/jpeg",
+			"originalName": "Fender Custom Shop 52 Tele (10 of 12).jpeg",
+			"originalUrl": "https://arc-anglerfish-local-localhost.s3.amazonaws.com/public/2K2QZFZXT5BPBPGFPKJIGL7D5Q.jpeg",
+			"owner": "test.user@arcpublishing.com",
+			"proxyUrl": "/photo/resize/pUlgam_6M2eiyWdw1xNDsOLjkuA=/fit-in/330x230/arc-anglerfish-local-localhost/public/2K2QZFZXT5BPBPGFPKJIGL7D5Q.jpeg",
+			"published": true,
+			"resizeUrl": "http://resizer.shared.arcpublishing.com/pUlgam_6M2eiyWdw1xNDsOLjkuA=/fit-in/330x230/arc-anglerfish-local-localhost/public/2K2QZFZXT5BPBPGFPKJIGL7D5Q.jpeg",
+			"restricted": false,
+			"takenOn": "2019-05-10T13:20:35Z",
+			"version": 33
+		},
+		"address": {},
+		"created_date": "2019-06-03T20:18:53Z",
+		"credits": {
+			"by": []
+		},
+    "focal_point": {
+      "x": 1146,
+      "y": 806
+    },
+		"geo": {},
+		"height": 4302,
+		"image_type": "illustration",
+		"last_updated_date": "2019-06-20T20:59:02Z",
+		"licensable": false,
+		"owner": {
+			"id": "localhost",
+			"sponsored": true
+		},
+		"source": {
+			"edit_url": "https://localhost.arcpublishing.com/photo/2K2QZFZXT5BPBPGFPKJIGL7D5Q",
+			"system": "Anglerfish"
+		},
+		"status": "true",
+		"taxonomy": {
+			"associated_tasks": []
+		},
+		"type": "image",
+		"url": "https://arc-anglerfish-local-localhost.s3.amazonaws.com/public/2K2QZFZXT5BPBPGFPKJIGL7D5Q.jpeg",
+		"version": "0.9.0",
+		"width": 2868,
+		"syndication": {
+			"search": false,
+			"external_distribution": false
+		}
+	}]
+}
+```
+
+### Example story ANS overriding the included image's focal point using referent_properties
+```
+{
+  "_id": "25NZ3GMIYJFKNNT5CNM36ZMKEA",
+  "type": "story",
+  "version": "0.10.2",
+  "content_elements": [
+    {
+      "type": "promo_reference",
+      "referent": {
+        "id": "67AWVTPPTFBHZD4ACJRU7FS4L4",
+        "type": "image",
+        "referent_properties": {
+          "focal_point": {
+              "x": 600,
+              "y": 400
+          },
+        }
+      }
+    }
+  ],
+  "created_date": "2019-07-11T20:29:38.908Z",
+  "last_updated_date": "2019-07-12T15:05:19.631Z",
+  "canonical_url": "\/about-us\/test-slug-20190711-25nz3gmiyjfknnt5cnm36zmkea-story.html",
+  "headlines": {
+    "basic": "Testing adding content element"
+  },
+  "owner": {
+    "sponsored": false,
+    "name": "The Washington Post",
+    "id": "staging"
+  },
+  "workflow": {
+    "status_code": 1
+  },
+  "source": {
+    "system": "ellipsis",
+    "name": "wp",
+    "source_type": "staff"
+  },
+  "additional_properties": {
+    "has_published_copy": true,
+    "is_published": true,
+    "publish_date": "2019-07-11T21:52:06.790Z"
+  }
+}
+```
+
+## Concerns
+No concerns at this time.
```

Comment:
Has anyone tried using this proposed ANS data format with a resizer (e.g. Thumbor) to actually create images with a different size/aspect ratio that use this focal point? That would be my only request/concern – make sure it's reasonably compatible with the main use cases. 

For example, let's say there's an Anglerfish image that's 3000x2000 px wide (note this is a 3x2 ratio). The focal point is set at 1000x1000. The website needs to render a version of this image in a 1x1 ratio, centering around the focal point. It needs to have a few different sizes of the image in order to serve the right size to desktop, tablet, and mobile users (let's say 600x600, 300x300, and 100x100 px images, respectively). 

Is the data we're proposing including in the ANS here (the focal point as an X, Y coordinate in pixels on the original image) enough information for a front-end developer to create different crops of the original image – centering on the focal point – with different aspect ratios and pixel sizes? 

If yes, then I think this is good. I would just like someone to confirm we think this use case can be handled adequately with this data.

Reply:
I second these concerns. When I look at the docs for the resizer docs in the ALC it looks like the front end will be expected to calculate and return the new dimensions for the image but it's not clear how this is supposed to be determined. Should the focal point always be 0,0 in the center or on a specific corner of the photo. https://rtl.arcpublishing.com/alc/arc-products/ellipsis/user-docs/ellipsis-using-focal-points-set-in-stories-with-the-resizer/

Reply:
This is sort of what I was getting at earlier, but my feeling based on the response was that this should be, by design, explicitly unknown client-side rendering details and outside of the purview of ANS. Different clients will have different rendering conditions and may use different resizes, and will make a call on their end.